packages/shared/src/write-archive/dom-snapshot.ts

```diff
@@ -103,6 +104,11 @@ export class DOMSnapshot {
       if (node.tagName === 'picture') {
         this.mapPictureElement(node, sourceMap);
       }
+
+      // If the element is a base tag, we remove the localhost from the href
```

Comment:
Since the base href value comes from upstream DOM snapshot - this is the best spot I can find to modify it by removing any localhost if there is any.